dds_web/api/schemas/project_schemas.py

```diff
@@ -0,0 +1,86 @@
+"""Project related marshmallow schemas."""
+
+####################################################################################################
+# IMPORTS ################################################################################ IMPORTS #
+####################################################################################################
+
+# Standard Library
+from datetime import datetime
+
+# Installed
+import flask
+import marshmallow
+import sqlalchemy
+
+# Own modules
+from dds_web.api import errors as ddserr
+from dds_web import auth
+from dds_web.database import models
+
+
+####################################################################################################
+# VALIDATORS ########################################################################## VALIDATORS #
+####################################################################################################
+
+
+def verify_project_exists(spec_proj):
+    """Check that project exists."""
+
+    try:
+        project = models.Project.query.filter(
+            models.Project.public_id == sqlalchemy.func.binary(spec_proj)
+        ).one_or_none()
+    except sqlalchemy.exc.SQLAlchemyError as sqlerr:
+        raise
+
+    if not project:
+        flask.current_app.logger.warning("No such project!!")
+        raise ddserr.NoSuchProjectError(project=spec_proj)
+
+    return project
+
+
+def verify_project_access(project):
+    """Check users access to project."""
+
+    if project not in auth.current_user().projects:
+        raise ddserr.AccessDeniedError(
+            message="Project access denied.",
+            username=auth.current_user().username,
+            project=project.public_id,
+        )
+
+    return project
+
+
+####################################################################################################
+# SCHEMAS ################################################################################ SCHEMAS #
+####################################################################################################
+
+
+class ProjectRequiredSchema(marshmallow.Schema):
+    """Schema for verifying an existing project in args and database."""
+
+    project = marshmallow.fields.String(required=True)
+
+    class Meta:
+        unknown = marshmallow.EXCLUDE  # TODO: Change to RAISE
```

Comment:
In some of the endpoints the `ProjectRequiredSchema` is called with `ProjectRequiredSchema().load(flask.request.args)` --> if there are other valid, but not used for this specific schema, args passed in it will raise a validation error which it shouldn't. The comment about changing back is for after the refactoring, then this shouldn't still happen.